vendor/openfx/OCIOColorSpace.cpp

```diff
@@ -0,0 +1,224 @@
+        identityTime = args.time;
+        return true;
+    }
+
+    return false;
+}
+
+void OCIOColorSpace::changedParam(const OFX::InstanceChangedArgs & /*args*/, 
+                                  const std::string & paramName)
+{
+    if (paramName == "src_cs" || paramName == "dst_cs")
+    {
+        OCIO::ConstConfigRcPtr config = getOCIOConfig();
+
+        std::string configName(config->getName());
+        if (!configName.empty())
+        {
+            configName = " '" + configName + "'";
+        }
+
+        std::string defaultViewTrName(config->getDefaultViewTransformName());
+        int numViewTransforms = config->getNumViewTransforms();
+
+        std::string srcCsName = getChoiceParamOption(srcCsNameParam_);
+        OCIO::ConstColorSpaceRcPtr srcCs = 
+            config->getColorSpace(srcCsName.c_str());
+        OCIO::ReferenceSpaceType srcRef = srcCs->getReferenceSpaceType();
+
+        std::string dstCsName = getChoiceParamOption(dstCsNameParam_);
+        OCIO::ConstColorSpaceRcPtr dstCs = 
+            config->getColorSpace(dstCsName.c_str());
+        OCIO::ReferenceSpaceType dstRef = dstCs->getReferenceSpaceType();
+
+        // Suggest using OCIODisplay instead
+        if (numViewTransforms > 1 && srcRef != dstRef)
+        {
+            std::ostringstream os;
+            os << "Color space '" << srcCsName << "' is ";
+            os << (srcRef == OCIO::REFERENCE_SPACE_SCENE ? "scene" : "display");
+            os << " referred and '" << dstCsName << "' is ";
+            os << (dstRef == OCIO::REFERENCE_SPACE_SCENE ? "scene" : "display");
+            os << " referred. The OCIO config" << configName << " contains ";
+            os << numViewTransforms << " view transforms and the default '";
+            os << defaultViewTrName << "' will be used for this conversion, ";
+            os << "which may not be the intended behvior. Please use ";
+            os << "'OCIODisplay' to ensure a correct viewing pipeline result.";
+
+            sendMessage(OFX::Message::eMessageWarning,
+                        "view_transform_warning",
+                        os.str());
+        }
+    }
+
+    else if (paramName == "swap_src_dst")
+    {
+        int srcCsIdx, dstCsIdx;
+        srcCsNameParam_->getValue(srcCsIdx);
+        dstCsNameParam_->getValue(dstCsIdx);
+
+        // Swap src and dst color space indices
+        srcCsNameParam_->setValue(dstCsIdx);
+        dstCsNameParam_->setValue(srcCsIdx);
+    }
+    else
+    {
+        contextParamChanged(*this, paramName);
+    }
+}
+
+void OCIOColorSpaceFactory::describe(OFX::ImageEffectDescriptor& desc)
+{
+    // Labels
+    desc.setLabels("OCIOColorSpace", "OCIOColorSpace", "OCIOColorSpace");
+    desc.setPluginGrouping("OpenColorIO");
+
+    // Supported contexts
+    desc.addSupportedContext(OFX::eContextFilter);
+    desc.addSupportedContext(OFX::eContextGeneral);
+
+    // Supported pixel depths
+    desc.addSupportedBitDepth(OFX::eBitDepthUByte);
```

Comment:
OFX supports uint8 uint16, float16, and float32 via an enum. It does have a special value for a "custom" bit-depth, but I have not investigated how to handle that yet. It would appear however that Flame is converting the 10 and 12 bit inputs to float to comply with the plugin's declared support for these four bit-depths, which in reading the OFX docs, is correct host behavior. The plugin is also currently declaring that it doesn't support multiple bit-depths, and I'm creating an optimized CPU processor with matching input and output depths. Perhaps I should drop support of the integer bit depths in the plugin declaration, which would enforce float processing, and leave it up to the host to convert that back to an integer if needed. I know that would limit the integer input/output optimization in OCIO, but it would also guarantee color space precision from of the plugin.

Reply:
I would guess that most of the usage of these plug-ins will have float on one side or the other or both.  Also, most hosts probably either don't support kOfxImageEffectPropSupportsMultipleClipDepths or don't give users easy control for managing the depths.  So I would concur that we should probably just limit support to halfs and floats.  This will be a performance hit for apps that might like to pass in integers but we will typically not want to return an integer anyway.

Reply:
Sounds good. Changed as discussed.